### TEAM WASHINGTON LP 1

# Questions

1. How many start-ups got funded each year?
2. In which year did start-ups receive the most funding?
3. What are the top ten start-ups that received the most funding?
4. What are the top ten start-ups that received the least funding?
5. What are the top ten start-up industries that received the most funding?
6. What are the top ten start-up industries that received the least funding?
7. What are the top ten start-up locations that received the most funding?
8. What are the top ten start-up locations that received the least funding?

In [232]:
# Importing libraries

import pandas as pd
import numpy as np
import matplotlib as mp
import seaborn as sns

In [233]:
# Loading the datasets

df1 = pd.read_csv('startup_funding2018.csv')
df2 = pd.read_csv('startup_funding2019.csv')
df3 = pd.read_csv('startup_funding2020.csv')
df4 = pd.read_csv('startup_funding2021.csv')

In [234]:
# Evaluating the first dataset

df1.head()

Company Name                                           Industry   
0  TheCollegeFever  Brand Marketing, Event Promotion, Marketing, S...  \
1  Happy Cow Dairy                               Agriculture, Farming   
2       MyLoanCare   Credit, Financial Services, Lending, Marketplace   
3      PayMe India                        Financial Services, FinTech   
4         Eunimart                 E-Commerce Platforms, Retail, SaaS   

  Round/Series       Amount                          Location   
0         Seed       250000       Bangalore, Karnataka, India  \
1         Seed  ₹40,000,000        Mumbai, Maharashtra, India   
2     Series A  ₹65,000,000           Gurgaon, Haryana, India   
3        Angel      2000000       Noida, Uttar Pradesh, India   
4         Seed            —  Hyderabad, Andhra Pradesh, India   

                                       About Company  
0  TheCollegeFever is a hub for fun, fiesta and f...  
1  A startup which aggregates milk from dairy far...  
2          Leading Online Loans Marketplace in India  
3  PayMe India is an innovative FinTech organizat...  
4  Eunimart is a one stop solution for merchants ...

In [235]:
# Evaluating the second dataset

df2.head()

Company/Brand  Founded HeadQuarter           Sector   
0  Bombay Shaving      NaN         NaN        Ecommerce  \
1       Ruangguru   2014.0      Mumbai           Edtech   
2        Eduisfun      NaN      Mumbai           Edtech   
3        HomeLane   2014.0     Chennai  Interior design   
4        Nu Genes   2004.0   Telangana         AgriTech   

                                        What it does   
0         Provides a range of male grooming products  \
1  A learning platform that provides topic-based ...   
2            It aims to make learning fun via games.   
3              Provides interior designing solutions   
4  It is a seed company engaged in production, pr...   

                                Founders   
0                     Shantanu Deshpande  \
1  Adamas Belva Syah Devara, Iman Usman.   
2                          Jatin Solanki   
3           Srikanth Iyer, Rama Harinath   
4                 Narayana Reddy Punyala   

                                            Investor     Amount($)   
0                               Sixth Sense Ventures    $6,300,000  \
1                                   General Atlantic  $150,000,000   
2     Deepak Parekh, Amitabh Bachchan, Piyush Pandey   $28,000,000   
3  Evolvence India Fund (EIF), Pidilite Group, FJ...   $30,000,000   
4           Innovation in Food and Agriculture (IFA)    $6,000,000   

           Stage  
0            NaN  
1       Series C  
2  Fresh funding  
3       Series D  
4            NaN

In [236]:
# Evaluating the third dataset

df3.head()

Company/Brand Founded HeadQuarter              Sector   
0    Aqgromalin    2019     Chennai            AgriTech  \
1      Krayonnz    2019   Bangalore              EdTech   
2  PadCare Labs    2018        Pune  Hygiene management   
3         NCOME    2020   New Delhi              Escrow   
4    Gramophone    2016      Indore            AgriTech   

                                        What it does   
0                       Cultivating Ideas for Profit  \
1  An academy-guardian-scholar centric ecosystem ...   
2   Converting bio-hazardous waste to harmless waste   
3                       Escrow-as-a-service platform   
4  Gramophone is an AgTech platform enabling acce...   

                                            Founders   
0                    Prasanna Manogaran, Bharani C L  \
1                   Saurabh Dixit, Gurudutt Upadhyay   
2                                    Ajinkya Dhariya   
3                                      Ritesh Tiwari   
4  Ashish Rajan Singh, Harshit Gupta, Nishant Mah...   

                              Investor    Amount($)     Stage Unnamed: 9  
0                      Angel investors     $200,000       NaN        NaN  
1                      GSF Accelerator     $100,000  Pre-seed        NaN  
2                       Venture Center  Undisclosed  Pre-seed        NaN  
3  Venture Catalysts, PointOne Capital     $400,000       NaN        NaN  
4  Siana Capital Management, Info Edge     $340,000       NaN        NaN

In [237]:
# Evaluating the fourth dataset

df4.head()

Company/Brand  Founded HeadQuarter          Sector   
0  Unbox Robotics   2019.0   Bangalore      AI startup  \
1          upGrad   2015.0      Mumbai          EdTech   
2     Lead School   2012.0      Mumbai          EdTech   
3         Bizongo   2015.0      Mumbai  B2B E-commerce   
4        FypMoney   2021.0    Gurugram         FinTech   

                                        What it does   
0  Unbox Robotics builds on-demand AI-driven ware...  \
1     UpGrad is an online higher education platform.   
2  LEAD School offers technology based school tra...   
3  Bizongo is a business-to-business online marke...   
4  FypMoney is Digital NEO Bank for Teenagers, em...   

                                            Founders   
0                        Pramod Ghadge, Shahid Memon  \
1  Mayank Kumar, Phalgun Kompalli, Ravijot Chugh,...   
2                         Smita Deorah, Sumeet Mehta   
3            Aniket Deb, Ankit Tomar, Sachin Agrawal   
4                                      Kapil Banwari   

                                        Investor     Amount($)         Stage  
0                    BEENEXT, Entrepreneur First    $1,200,000  Pre-series A  
1       Unilazer Ventures, IIFL Asset Management  $120,000,000           NaN  
2               GSV Ventures, Westbridge Capital   $30,000,000      Series D  
3                         CDC Group, IDG Capital   $51,000,000      Series C  
4  Liberatha Kallat, Mukesh Yadav, Dinesh Nagpal    $2,000,000          Seed

# Observations

The first dataset does not have columns named "Founded", "Founders" and "Investor" which are in the other three datasets. Considering the fact that the information on these columns do not affect our hypothesis and questions, it is likely that those columns will be dropped (unless there is a way to get the accurate information for those columns) to enable us merge all the datasets later. We will do this while cleaning the datasets.

Also, there is a column named "Unnamed: 9" in the third dataset. Its first five rows have missing values, and other datasets do not contain such column. This column will be examined while cleaning the third dataset.

Furthermore, all the datasets have similar column names except the first dataset. This means that there is need to edit the column names of the first dataset to match the column names of the other datasets. This will be done while cleaning the first dataset.

Finally, there will be need to add a column named "Year" to each of the datasets. This year column will show the 'dataset year' that each row belong to for easy identification when all the datasets are merged, as this will help us answer our questions.

# Cleaning the first dataset

In [238]:
# Evaluating the dataset

df1.head()

Company Name                                           Industry   
0  TheCollegeFever  Brand Marketing, Event Promotion, Marketing, S...  \
1  Happy Cow Dairy                               Agriculture, Farming   
2       MyLoanCare   Credit, Financial Services, Lending, Marketplace   
3      PayMe India                        Financial Services, FinTech   
4         Eunimart                 E-Commerce Platforms, Retail, SaaS   

  Round/Series       Amount                          Location   
0         Seed       250000       Bangalore, Karnataka, India  \
1         Seed  ₹40,000,000        Mumbai, Maharashtra, India   
2     Series A  ₹65,000,000           Gurgaon, Haryana, India   
3        Angel      2000000       Noida, Uttar Pradesh, India   
4         Seed            —  Hyderabad, Andhra Pradesh, India   

                                       About Company  
0  TheCollegeFever is a hub for fun, fiesta and f...  
1  A startup which aggregates milk from dairy far...  
2          Leading Online Loans Marketplace in India  
3  PayMe India is an innovative FinTech organizat...  
4  Eunimart is a one stop solution for merchants ...

In [239]:
# Changing the column names using a dictionary

column_dict = {
    'Company Name': 'Company/Brand',
    'Industry': 'Sector',
    'Round/Series': 'Stage',
    'Location': 'HeadQuarter',
    'About Company': 'What it does',
    'Amount': 'Amount($)'
}

df1 = df1.rename(columns=column_dict)
df1.head()

Company/Brand                                             Sector   
0  TheCollegeFever  Brand Marketing, Event Promotion, Marketing, S...  \
1  Happy Cow Dairy                               Agriculture, Farming   
2       MyLoanCare   Credit, Financial Services, Lending, Marketplace   
3      PayMe India                        Financial Services, FinTech   
4         Eunimart                 E-Commerce Platforms, Retail, SaaS   

      Stage    Amount($)                       HeadQuarter   
0      Seed       250000       Bangalore, Karnataka, India  \
1      Seed  ₹40,000,000        Mumbai, Maharashtra, India   
2  Series A  ₹65,000,000           Gurgaon, Haryana, India   
3     Angel      2000000       Noida, Uttar Pradesh, India   
4      Seed            —  Hyderabad, Andhra Pradesh, India   

                                        What it does  
0  TheCollegeFever is a hub for fun, fiesta and f...  
1  A startup which aggregates milk from dairy far...  
2          Leading Online Loans Marketplace in India  
3  PayMe India is an innovative FinTech organizat...  
4  Eunimart is a one stop solution for merchants ...

In [240]:
df1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 526 entries, 0 to 525
Data columns (total 6 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   Company/Brand  526 non-null    object
 1   Sector         526 non-null    object
 2   Stage          526 non-null    object
 3   Amount($)      526 non-null    object
 4   HeadQuarter    526 non-null    object
 5   What it does   526 non-null    object
dtypes: object(6)
memory usage: 24.8+ KB


In [241]:
# Confirming that there are no missing values

df1.isna().sum()

Company/Brand    0
Sector           0
Stage            0
Amount($)        0
HeadQuarter      0
What it does     0
dtype: int64

There are no missing values in the dataset.

In [242]:
# Checking for duplicate rows

df1.duplicated().sum()

1

The dataset has one duplicate row

In [243]:
# Dropping the duplicate row

df1 = df1.drop_duplicates()

In [244]:
# Confirming that the duplicate row has been dropped

df1.duplicated().sum()

0

The duplicate row has been dropped

In [245]:
df1.info()

<class 'pandas.core.frame.DataFrame'>
Index: 525 entries, 0 to 525
Data columns (total 6 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   Company/Brand  525 non-null    object
 1   Sector         525 non-null    object
 2   Stage          525 non-null    object
 3   Amount($)      525 non-null    object
 4   HeadQuarter    525 non-null    object
 5   What it does   525 non-null    object
dtypes: object(6)
memory usage: 28.7+ KB


After dropping the duplicate row, the number of rows reduced from 526 to 525

# Checking for custom quality
The cells of the "Amounts ($)" colummn of the dataset has amounts in rupees, amounts in dollars, and dash symbols. The rupee amounts will be converted to dollars since the amounts in other datasets are also in dollars. The average exchange rate of rupees to dollars in 2018 will be used to ensure accurate conversion. The dash symbols will be handled afterwards.

In [246]:
# Before converting, let's check the datatypes of the values in the "Amount($)" column

df1["Amount($)"].apply(type).value_counts()

Amount($)
<class 'str'>    525
Name: count, dtype: int64

The rupee amounts alone need to be converted and the presence of the rupee symbol in the amounts suggest that the datatype would be string. There is need to change the datatypes to float to make the conversion possible. We will first remove all the non-numerical values then applying .astype(float). The non-numerical values are the rupee symbol and the comma symbol. While removing them, we will use the lambda function to ensure that NaN values are not returned.

In [247]:
# Define the average exchange rate in 2018
exchange_rate = 0.0146  # That is, 1 Rupee = 0.0146 Dollar

# Select the rows with rupee amounts
rupee_rows = df1["Amount($)"].str.contains('₹') # Note that only the rupee amounts should be converted 

# Remove the non-numerical values and change the datatype to float
df1.loc[rupee_rows, "Amount($)"] = df1.loc[rupee_rows, "Amount($)"].apply(lambda x: x.replace('₹', '').replace(',', '')
                                if isinstance(x, str) else x).astype(float)

# Convert the rupee amounts to dollars
df1.loc[rupee_rows, "Amount($)"] = df1.loc[rupee_rows, "Amount($)"] * exchange_rate

df1

Company/Brand                                             Sector   
0      TheCollegeFever  Brand Marketing, Event Promotion, Marketing, S...  \
1      Happy Cow Dairy                               Agriculture, Farming   
2           MyLoanCare   Credit, Financial Services, Lending, Marketplace   
3          PayMe India                        Financial Services, FinTech   
4             Eunimart                 E-Commerce Platforms, Retail, SaaS   
..                 ...                                                ...   
521              Udaan   B2B, Business Development, Internet, Marketplace   
522  Happyeasygo Group                                    Tourism, Travel   
523             Mombay         Food and Beverage, Food Delivery, Internet   
524         Droni Tech                             Information Technology   
525            Netmeds         Biotechnology, Health Care, Pharmaceutical   

        Stage  Amount($)                       HeadQuarter   
0        Seed     250000       Bangalore, Karnataka, India  \
1        Seed   584000.0        Mumbai, Maharashtra, India   
2    Series A   949000.0           Gurgaon, Haryana, India   
3       Angel    2000000       Noida, Uttar Pradesh, India   
4        Seed          —  Hyderabad, Andhra Pradesh, India   
..        ...        ...                               ...   
521  Series C  225000000       Bangalore, Karnataka, India   
522  Series A          —           Haryana, Haryana, India   
523      Seed       7500        Mumbai, Maharashtra, India   
524      Seed   511000.0        Mumbai, Maharashtra, India   
525  Series C   35000000        Chennai, Tamil Nadu, India   

                                          What it does  
0    TheCollegeFever is a hub for fun, fiesta and f...  
1    A startup which aggregates milk from dairy far...  
2            Leading Online Loans Marketplace in India  
3    PayMe India is an innovative FinTech organizat...  
4    Eunimart is a one stop solution for merchants ...  
..                                                 ...  
521  Udaan is a B2B trade platform, designed specif...  
522            HappyEasyGo is an online travel domain.  
523  Mombay is a unique opportunity for housewives ...  
524  Droni Tech manufacture UAVs and develop softwa...  
525       Welcome to India's most convenient pharmacy!  

[525 rows x 6 columns]

# Handling the dash symbols

In [248]:
# Counting the number of cells containing the dash symbol

(df1["Amount($)"] == '—').sum()

148

The dash symbol in the amount column of 148 startups could mean that the start-ups received no funding, or that the actual data were lost during the data collection process. Since the dataset originally did not contain missing values, assuming that some rows have dash symbols because the actual data was lost during the data collection process will be unnecessary. It is safer to assume that the dash sumbol represent zero fuding. For this reason, the dash symbols will be replaced with zero (0).

In [249]:
# Replacing the dash symbol with zero

df1['Amount($)'] = df1['Amount($)'].replace('—', '0', regex=True)
df1

Company/Brand                                             Sector   
0      TheCollegeFever  Brand Marketing, Event Promotion, Marketing, S...  \
1      Happy Cow Dairy                               Agriculture, Farming   
2           MyLoanCare   Credit, Financial Services, Lending, Marketplace   
3          PayMe India                        Financial Services, FinTech   
4             Eunimart                 E-Commerce Platforms, Retail, SaaS   
..                 ...                                                ...   
521              Udaan   B2B, Business Development, Internet, Marketplace   
522  Happyeasygo Group                                    Tourism, Travel   
523             Mombay         Food and Beverage, Food Delivery, Internet   
524         Droni Tech                             Information Technology   
525            Netmeds         Biotechnology, Health Care, Pharmaceutical   

        Stage  Amount($)                       HeadQuarter   
0        Seed     250000       Bangalore, Karnataka, India  \
1        Seed   584000.0        Mumbai, Maharashtra, India   
2    Series A   949000.0           Gurgaon, Haryana, India   
3       Angel    2000000       Noida, Uttar Pradesh, India   
4        Seed          0  Hyderabad, Andhra Pradesh, India   
..        ...        ...                               ...   
521  Series C  225000000       Bangalore, Karnataka, India   
522  Series A          0           Haryana, Haryana, India   
523      Seed       7500        Mumbai, Maharashtra, India   
524      Seed   511000.0        Mumbai, Maharashtra, India   
525  Series C   35000000        Chennai, Tamil Nadu, India   

                                          What it does  
0    TheCollegeFever is a hub for fun, fiesta and f...  
1    A startup which aggregates milk from dairy far...  
2            Leading Online Loans Marketplace in India  
3    PayMe India is an innovative FinTech organizat...  
4    Eunimart is a one stop solution for merchants ...  
..                                                 ...  
521  Udaan is a B2B trade platform, designed specif...  
522            HappyEasyGo is an online travel domain.  
523  Mombay is a unique opportunity for housewives ...  
524  Droni Tech manufacture UAVs and develop softwa...  
525       Welcome to India's most convenient pharmacy!  

[525 rows x 6 columns]

In [250]:
# Removing possible dollar and commma symbols in the entire Amount($) column

df1['Amount($)'] = df1['Amount($)'].replace(['\$', ','], '', regex=True)
df1

Company/Brand                                             Sector   
0      TheCollegeFever  Brand Marketing, Event Promotion, Marketing, S...  \
1      Happy Cow Dairy                               Agriculture, Farming   
2           MyLoanCare   Credit, Financial Services, Lending, Marketplace   
3          PayMe India                        Financial Services, FinTech   
4             Eunimart                 E-Commerce Platforms, Retail, SaaS   
..                 ...                                                ...   
521              Udaan   B2B, Business Development, Internet, Marketplace   
522  Happyeasygo Group                                    Tourism, Travel   
523             Mombay         Food and Beverage, Food Delivery, Internet   
524         Droni Tech                             Information Technology   
525            Netmeds         Biotechnology, Health Care, Pharmaceutical   

        Stage  Amount($)                       HeadQuarter   
0        Seed     250000       Bangalore, Karnataka, India  \
1        Seed   584000.0        Mumbai, Maharashtra, India   
2    Series A   949000.0           Gurgaon, Haryana, India   
3       Angel    2000000       Noida, Uttar Pradesh, India   
4        Seed          0  Hyderabad, Andhra Pradesh, India   
..        ...        ...                               ...   
521  Series C  225000000       Bangalore, Karnataka, India   
522  Series A          0           Haryana, Haryana, India   
523      Seed       7500        Mumbai, Maharashtra, India   
524      Seed   511000.0        Mumbai, Maharashtra, India   
525  Series C   35000000        Chennai, Tamil Nadu, India   

                                          What it does  
0    TheCollegeFever is a hub for fun, fiesta and f...  
1    A startup which aggregates milk from dairy far...  
2            Leading Online Loans Marketplace in India  
3    PayMe India is an innovative FinTech organizat...  
4    Eunimart is a one stop solution for merchants ...  
..                                                 ...  
521  Udaan is a B2B trade platform, designed specif...  
522            HappyEasyGo is an online travel domain.  
523  Mombay is a unique opportunity for housewives ...  
524  Droni Tech manufacture UAVs and develop softwa...  
525       Welcome to India's most convenient pharmacy!  

[525 rows x 6 columns]

In [251]:
# Changing the datatype of the Amount($) column back to integer

df1["Amount($)"] = df1["Amount($)"].astype(np.int64)

df1.head()

Company/Brand                                             Sector   
0  TheCollegeFever  Brand Marketing, Event Promotion, Marketing, S...  \
1  Happy Cow Dairy                               Agriculture, Farming   
2       MyLoanCare   Credit, Financial Services, Lending, Marketplace   
3      PayMe India                        Financial Services, FinTech   
4         Eunimart                 E-Commerce Platforms, Retail, SaaS   

      Stage  Amount($)                       HeadQuarter   
0      Seed     250000       Bangalore, Karnataka, India  \
1      Seed     584000        Mumbai, Maharashtra, India   
2  Series A     949000           Gurgaon, Haryana, India   
3     Angel    2000000       Noida, Uttar Pradesh, India   
4      Seed          0  Hyderabad, Andhra Pradesh, India   

                                        What it does  
0  TheCollegeFever is a hub for fun, fiesta and f...  
1  A startup which aggregates milk from dairy far...  
2          Leading Online Loans Marketplace in India  
3  PayMe India is an innovative FinTech organizat...  
4  Eunimart is a one stop solution for merchants ...

In [252]:
# Rearranging the columns in the dataset to fit the format in the second, third and fourth datasets

df1 = df1.loc[:, ["Company/Brand", "HeadQuarter", "Sector", "What it does", "Amount($)", "Stage"]]
df1.head()

Company/Brand                       HeadQuarter   
0  TheCollegeFever       Bangalore, Karnataka, India  \
1  Happy Cow Dairy        Mumbai, Maharashtra, India   
2       MyLoanCare           Gurgaon, Haryana, India   
3      PayMe India       Noida, Uttar Pradesh, India   
4         Eunimart  Hyderabad, Andhra Pradesh, India   

                                              Sector   
0  Brand Marketing, Event Promotion, Marketing, S...  \
1                               Agriculture, Farming   
2   Credit, Financial Services, Lending, Marketplace   
3                        Financial Services, FinTech   
4                 E-Commerce Platforms, Retail, SaaS   

                                        What it does  Amount($)     Stage  
0  TheCollegeFever is a hub for fun, fiesta and f...     250000      Seed  
1  A startup which aggregates milk from dairy far...     584000      Seed  
2          Leading Online Loans Marketplace in India     949000  Series A  
3  PayMe India is an innovative FinTech organizat...    2000000     Angel  
4  Eunimart is a one stop solution for merchants ...          0      Seed

In the first dataset, the "Headquarter" column contains the names of the cities, states and country. But the other datasets have only  the names of cities in the "Headquarter" column. In order to ensure that the datasets maintain the same format, the names of states and country will be removed from the "Headquarter" column of the first dataset, leaving only the names of cities.

In [253]:
# Split the column values by comma and select only the first word (names of cities)

df1["HeadQuarter"] = df1["HeadQuarter"].str.split(',').str[0]
df1

Company/Brand HeadQuarter   
0      TheCollegeFever   Bangalore  \
1      Happy Cow Dairy      Mumbai   
2           MyLoanCare     Gurgaon   
3          PayMe India       Noida   
4             Eunimart   Hyderabad   
..                 ...         ...   
521              Udaan   Bangalore   
522  Happyeasygo Group     Haryana   
523             Mombay      Mumbai   
524         Droni Tech      Mumbai   
525            Netmeds     Chennai   

                                                Sector   
0    Brand Marketing, Event Promotion, Marketing, S...  \
1                                 Agriculture, Farming   
2     Credit, Financial Services, Lending, Marketplace   
3                          Financial Services, FinTech   
4                   E-Commerce Platforms, Retail, SaaS   
..                                                 ...   
521   B2B, Business Development, Internet, Marketplace   
522                                    Tourism, Travel   
523         Food and Beverage, Food Delivery, Internet   
524                             Information Technology   
525         Biotechnology, Health Care, Pharmaceutical   

                                          What it does  Amount($)     Stage  
0    TheCollegeFever is a hub for fun, fiesta and f...     250000      Seed  
1    A startup which aggregates milk from dairy far...     584000      Seed  
2            Leading Online Loans Marketplace in India     949000  Series A  
3    PayMe India is an innovative FinTech organizat...    2000000     Angel  
4    Eunimart is a one stop solution for merchants ...          0      Seed  
..                                                 ...        ...       ...  
521  Udaan is a B2B trade platform, designed specif...  225000000  Series C  
522            HappyEasyGo is an online travel domain.          0  Series A  
523  Mombay is a unique opportunity for housewives ...       7500      Seed  
524  Droni Tech manufacture UAVs and develop softwa...     511000      Seed  
525       Welcome to India's most convenient pharmacy!   35000000  Series C  

[525 rows x 6 columns]

# Undone work
The sector column in the first dataset does not have specific values, unlike the sector column of other datasets. What should we do about it as the sector column is an important focus area based on the hypothesis and questions

In [254]:
# Creating a new column named 'Year' and filling it with 2018, which is the dataset year

df1['Year'] = 2018
df1.head()

Company/Brand HeadQuarter   
0  TheCollegeFever   Bangalore  \
1  Happy Cow Dairy      Mumbai   
2       MyLoanCare     Gurgaon   
3      PayMe India       Noida   
4         Eunimart   Hyderabad   

                                              Sector   
0  Brand Marketing, Event Promotion, Marketing, S...  \
1                               Agriculture, Farming   
2   Credit, Financial Services, Lending, Marketplace   
3                        Financial Services, FinTech   
4                 E-Commerce Platforms, Retail, SaaS   

                                        What it does  Amount($)     Stage   
0  TheCollegeFever is a hub for fun, fiesta and f...     250000      Seed  \
1  A startup which aggregates milk from dairy far...     584000      Seed   
2          Leading Online Loans Marketplace in India     949000  Series A   
3  PayMe India is an innovative FinTech organizat...    2000000     Angel   
4  Eunimart is a one stop solution for merchants ...          0      Seed   

   Year  
0  2018  
1  2018  
2  2018  
3  2018  
4  2018

# Cleaning the second data set

In [255]:
# Evaluating the dataset

df2

Company/Brand  Founded HeadQuarter                   Sector   
0    Bombay Shaving      NaN         NaN                Ecommerce  \
1         Ruangguru   2014.0      Mumbai                   Edtech   
2          Eduisfun      NaN      Mumbai                   Edtech   
3          HomeLane   2014.0     Chennai          Interior design   
4          Nu Genes   2004.0   Telangana                 AgriTech   
..              ...      ...         ...                      ...   
84     Infra.Market      NaN      Mumbai                Infratech   
85              Oyo   2013.0    Gurugram              Hospitality   
86       GoMechanic   2016.0       Delhi  Automobile & Technology   
87           Spinny   2015.0       Delhi               Automobile   
88  Ess Kay Fincorp      NaN   Rajasthan                  Banking   

                                         What it does   
0          Provides a range of male grooming products  \
1   A learning platform that provides topic-based ...   
2             It aims to make learning fun via games.   
3               Provides interior designing solutions   
4   It is a seed company engaged in production, pr...   
..                                                ...   
84  It connects client requirements to their suppl...   
85                Provides rooms for comfortable stay   
86  Find automobile repair and maintenance service...   
87                                Online car retailer   
88              Organised Non-Banking Finance Company   

                                             Founders   
0                                  Shantanu Deshpande  \
1               Adamas Belva Syah Devara, Iman Usman.   
2                                       Jatin Solanki   
3                        Srikanth Iyer, Rama Harinath   
4                              Narayana Reddy Punyala   
..                                                ...   
84                    Aaditya Sharda, Souvik Sengupta   
85                                     Ritesh Agarwal   
86  Amit Bhasin, Kushal Karwa, Nitin Rana, Rishabh...   
87  Niraj Singh, Ramanshu Mahaur, Ganesh Pawar, Mo...   
88                                     Rajendra Setia   

                                             Investor     Amount($)   
0                                Sixth Sense Ventures    $6,300,000  \
1                                    General Atlantic  $150,000,000   
2      Deepak Parekh, Amitabh Bachchan, Piyush Pandey   $28,000,000   
3   Evolvence India Fund (EIF), Pidilite Group, FJ...   $30,000,000   
4            Innovation in Food and Agriculture (IFA)    $6,000,000   
..                                                ...           ...   
84  Tiger Global, Nexus Venture Partners, Accel Pa...   $20,000,000   
85  MyPreferred Transformation, Avendus Finance, S...  $693,000,000   
86                                    Sequoia Capital    $5,000,000   
87  Norwest Venture Partners, General Catalyst, Fu...   $50,000,000   
88     TPG, Norwest Venture Partners, Evolvence India   $33,000,000   

            Stage  
0             NaN  
1        Series C  
2   Fresh funding  
3        Series D  
4             NaN  
..            ...  
84       Series A  
85            NaN  
86       Series B  
87            NaN  
88            NaN  

[89 rows x 9 columns]

In [256]:
# Dropping the "Founded", "Founders" and "Investor" columns

df2 = df2.drop(["Founded", "Founders", "Investor"], axis =1)
df2.head()

Company/Brand HeadQuarter           Sector   
0  Bombay Shaving         NaN        Ecommerce  \
1       Ruangguru      Mumbai           Edtech   
2        Eduisfun      Mumbai           Edtech   
3        HomeLane     Chennai  Interior design   
4        Nu Genes   Telangana         AgriTech   

                                        What it does     Amount($)   
0         Provides a range of male grooming products    $6,300,000  \
1  A learning platform that provides topic-based ...  $150,000,000   
2            It aims to make learning fun via games.   $28,000,000   
3              Provides interior designing solutions   $30,000,000   
4  It is a seed company engaged in production, pr...    $6,000,000   

           Stage  
0            NaN  
1       Series C  
2  Fresh funding  
3       Series D  
4            NaN

In [257]:
# Checking for missing values

df2.isna().sum()

Company/Brand     0
HeadQuarter      19
Sector            5
What it does      0
Amount($)         0
Stage            46
dtype: int64

In [258]:
#checking the total number of rows

df2.shape

(89, 6)

Since there are only 89 rows in the dataset, dropping the rows with missing values will affect the end result of the analysis as only few rows will be left. As a result, we will use the backward fill method to fill the missing values

In [259]:
# Filling the missing values using backward fill method

df2["HeadQuarter"] = df2["HeadQuarter"].fillna(method="bfill")
df2["Sector"] = df2["Sector"].fillna(method="bfill")
df2["Stage"] = df2["Stage"].fillna(method="bfill")

df2.isna().sum()

Company/Brand    0
HeadQuarter      0
Sector           0
What it does     0
Amount($)        0
Stage            2
dtype: int64

In [260]:
# Dropping the remaining missing values

df2 = df2.dropna()

df2.isna().sum()

Company/Brand    0
HeadQuarter      0
Sector           0
What it does     0
Amount($)        0
Stage            0
dtype: int64

In [261]:
df2.head()

Company/Brand HeadQuarter           Sector   
0  Bombay Shaving      Mumbai        Ecommerce  \
1       Ruangguru      Mumbai           Edtech   
2        Eduisfun      Mumbai           Edtech   
3        HomeLane     Chennai  Interior design   
4        Nu Genes   Telangana         AgriTech   

                                        What it does     Amount($)   
0         Provides a range of male grooming products    $6,300,000  \
1  A learning platform that provides topic-based ...  $150,000,000   
2            It aims to make learning fun via games.   $28,000,000   
3              Provides interior designing solutions   $30,000,000   
4  It is a seed company engaged in production, pr...    $6,000,000   

           Stage  
0       Series C  
1       Series C  
2  Fresh funding  
3       Series D  
4   Pre series A

In [262]:
# Checking for duplicate rows

df2.duplicated().sum()

0

There are no duplicate rows

In [263]:
# Checking for custom quality

df2["Amount($)"].apply(type).value_counts()

Amount($)
<class 'str'>    87
Name: count, dtype: int64

In [264]:
# Removing the dollar and commma symbols in the entire Amount($) column

df2['Amount($)'] = df2['Amount($)'].replace(['\$', ','], '', regex=True)
df2.head()

Company/Brand HeadQuarter           Sector   
0  Bombay Shaving      Mumbai        Ecommerce  \
1       Ruangguru      Mumbai           Edtech   
2        Eduisfun      Mumbai           Edtech   
3        HomeLane     Chennai  Interior design   
4        Nu Genes   Telangana         AgriTech   

                                        What it does  Amount($)          Stage  
0         Provides a range of male grooming products    6300000       Series C  
1  A learning platform that provides topic-based ...  150000000       Series C  
2            It aims to make learning fun via games.   28000000  Fresh funding  
3              Provides interior designing solutions   30000000       Series D  
4  It is a seed company engaged in production, pr...    6000000   Pre series A

In [266]:
# Creating a new column named 'Year' and filling it with 2019, which is the dataset year

df2['Year'] = 2019
df2.head()

Company/Brand HeadQuarter           Sector   
0  Bombay Shaving      Mumbai        Ecommerce  \
1       Ruangguru      Mumbai           Edtech   
2        Eduisfun      Mumbai           Edtech   
3        HomeLane     Chennai  Interior design   
4        Nu Genes   Telangana         AgriTech   

                                        What it does  Amount($)   
0         Provides a range of male grooming products    6300000  \
1  A learning platform that provides topic-based ...  150000000   
2            It aims to make learning fun via games.   28000000   
3              Provides interior designing solutions   30000000   
4  It is a seed company engaged in production, pr...    6000000   

           Stage  Year  
0       Series C  2019  
1       Series C  2019  
2  Fresh funding  2019  
3       Series D  2019  
4   Pre series A  2019

# Cleaning the third data set

In [267]:
# Evaluating the third dataset

df3.head()

Company/Brand Founded HeadQuarter              Sector   
0    Aqgromalin    2019     Chennai            AgriTech  \
1      Krayonnz    2019   Bangalore              EdTech   
2  PadCare Labs    2018        Pune  Hygiene management   
3         NCOME    2020   New Delhi              Escrow   
4    Gramophone    2016      Indore            AgriTech   

                                        What it does   
0                       Cultivating Ideas for Profit  \
1  An academy-guardian-scholar centric ecosystem ...   
2   Converting bio-hazardous waste to harmless waste   
3                       Escrow-as-a-service platform   
4  Gramophone is an AgTech platform enabling acce...   

                                            Founders   
0                    Prasanna Manogaran, Bharani C L  \
1                   Saurabh Dixit, Gurudutt Upadhyay   
2                                    Ajinkya Dhariya   
3                                      Ritesh Tiwari   
4  Ashish Rajan Singh, Harshit Gupta, Nishant Mah...   

                              Investor    Amount($)     Stage Unnamed: 9  
0                      Angel investors     $200,000       NaN        NaN  
1                      GSF Accelerator     $100,000  Pre-seed        NaN  
2                       Venture Center  Undisclosed  Pre-seed        NaN  
3  Venture Catalysts, PointOne Capital     $400,000       NaN        NaN  
4  Siana Capital Management, Info Edge     $340,000       NaN        NaN

In [268]:
# Dropping the "Founded", "Founders" and "Investor" columns

df3 = df3.drop(["Founded", "Founders", "Investor"], axis =1)
df3.head()

Company/Brand HeadQuarter              Sector   
0    Aqgromalin     Chennai            AgriTech  \
1      Krayonnz   Bangalore              EdTech   
2  PadCare Labs        Pune  Hygiene management   
3         NCOME   New Delhi              Escrow   
4    Gramophone      Indore            AgriTech   

                                        What it does    Amount($)     Stage   
0                       Cultivating Ideas for Profit     $200,000       NaN  \
1  An academy-guardian-scholar centric ecosystem ...     $100,000  Pre-seed   
2   Converting bio-hazardous waste to harmless waste  Undisclosed  Pre-seed   
3                       Escrow-as-a-service platform     $400,000       NaN   
4  Gramophone is an AgTech platform enabling acce...     $340,000       NaN   

  Unnamed: 9  
0        NaN  
1        NaN  
2        NaN  
3        NaN  
4        NaN

In [269]:
df3.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1055 entries, 0 to 1054
Data columns (total 7 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   Company/Brand  1055 non-null   object
 1   HeadQuarter    961 non-null    object
 2   Sector         1042 non-null   object
 3   What it does   1055 non-null   object
 4   Amount($)      1052 non-null   object
 5   Stage          591 non-null    object
 6   Unnamed: 9     2 non-null      object
dtypes: object(7)
memory usage: 57.8+ KB


# Handling the "Unnamed: 9" column in the third dataset
Since the "Unnamed: 9" column contains only two non-null cells, it will be best to drop it.

In [270]:
# Dropping the "Unnamed: 9" column

df3 = df3.drop(["Unnamed: 9"], axis =1)
df3.head()

Company/Brand HeadQuarter              Sector   
0    Aqgromalin     Chennai            AgriTech  \
1      Krayonnz   Bangalore              EdTech   
2  PadCare Labs        Pune  Hygiene management   
3         NCOME   New Delhi              Escrow   
4    Gramophone      Indore            AgriTech   

                                        What it does    Amount($)     Stage  
0                       Cultivating Ideas for Profit     $200,000       NaN  
1  An academy-guardian-scholar centric ecosystem ...     $100,000  Pre-seed  
2   Converting bio-hazardous waste to harmless waste  Undisclosed  Pre-seed  
3                       Escrow-as-a-service platform     $400,000       NaN  
4  Gramophone is an AgTech platform enabling acce...     $340,000       NaN

In [271]:
# Checking for missing values

df3.isna().sum()

Company/Brand      0
HeadQuarter       94
Sector            13
What it does       0
Amount($)          3
Stage            464
dtype: int64

In [272]:
# Checking the total number of rows

df3.shape

(1055, 6)

Since there are 1055 rows in the dataset, we can drop the missing values in the "Sector" and "Amount ($)" columns since they are little, and only have to fill the missing values in the "HeadQuarter" and "Stage" columns since they are much.

In [273]:
# Filling the missing values using backward fill method

df3["HeadQuarter"] = df3["HeadQuarter"].fillna(method="bfill")
df3["Stage"] = df3["Stage"].fillna(method="bfill")

df3.isna().sum()

Company/Brand     0
HeadQuarter       0
Sector           13
What it does      0
Amount($)         3
Stage             0
dtype: int64

In [274]:
# Dropping the remaining missing values

df3 = df3.dropna()

df3.isna().sum()

Company/Brand    0
HeadQuarter      0
Sector           0
What it does     0
Amount($)        0
Stage            0
dtype: int64

In [275]:
df3.head()

Company/Brand HeadQuarter              Sector   
0    Aqgromalin     Chennai            AgriTech  \
1      Krayonnz   Bangalore              EdTech   
2  PadCare Labs        Pune  Hygiene management   
3         NCOME   New Delhi              Escrow   
4    Gramophone      Indore            AgriTech   

                                        What it does    Amount($)     Stage  
0                       Cultivating Ideas for Profit     $200,000  Pre-seed  
1  An academy-guardian-scholar centric ecosystem ...     $100,000  Pre-seed  
2   Converting bio-hazardous waste to harmless waste  Undisclosed  Pre-seed  
3                       Escrow-as-a-service platform     $400,000      Seed  
4  Gramophone is an AgTech platform enabling acce...     $340,000      Seed

In [276]:
# Checking for duplicate rows

df3.duplicated().sum()

2

The dataset has two duplicate rows

In [278]:
# Dropping the duplicate row

df3 = df3.drop_duplicates()

# Confirming that the duplicate row has been dropped

df3.duplicated().sum()

0

In [279]:
# Checking for custom quality

df3["Amount($)"].apply(type).value_counts()

Amount($)
<class 'str'>    1037
Name: count, dtype: int64

In [280]:
# Removing the dollar and commma symbols in the entire Amount($) column

df3['Amount($)'] = df3['Amount($)'].replace(['\$', ','], '', regex=True)
df3.head()

Company/Brand HeadQuarter              Sector   
0    Aqgromalin     Chennai            AgriTech  \
1      Krayonnz   Bangalore              EdTech   
2  PadCare Labs        Pune  Hygiene management   
3         NCOME   New Delhi              Escrow   
4    Gramophone      Indore            AgriTech   

                                        What it does    Amount($)     Stage  
0                       Cultivating Ideas for Profit       200000  Pre-seed  
1  An academy-guardian-scholar centric ecosystem ...       100000  Pre-seed  
2   Converting bio-hazardous waste to harmless waste  Undisclosed  Pre-seed  
3                       Escrow-as-a-service platform       400000      Seed  
4  Gramophone is an AgTech platform enabling acce...       340000      Seed

In [281]:
# Creating a new column named 'Year' and filling it with 2020, which is the dataset year

df3['Year'] = 2020
df3.head()

Company/Brand HeadQuarter              Sector   
0    Aqgromalin     Chennai            AgriTech  \
1      Krayonnz   Bangalore              EdTech   
2  PadCare Labs        Pune  Hygiene management   
3         NCOME   New Delhi              Escrow   
4    Gramophone      Indore            AgriTech   

                                        What it does    Amount($)     Stage   
0                       Cultivating Ideas for Profit       200000  Pre-seed  \
1  An academy-guardian-scholar centric ecosystem ...       100000  Pre-seed   
2   Converting bio-hazardous waste to harmless waste  Undisclosed  Pre-seed   
3                       Escrow-as-a-service platform       400000      Seed   
4  Gramophone is an AgTech platform enabling acce...       340000      Seed   

   Year  
0  2020  
1  2020  
2  2020  
3  2020  
4  2020

# Cleaning the fourth data set

In [282]:
# Evaluating the fourth dataset

df4.head()

Company/Brand  Founded HeadQuarter          Sector   
0  Unbox Robotics   2019.0   Bangalore      AI startup  \
1          upGrad   2015.0      Mumbai          EdTech   
2     Lead School   2012.0      Mumbai          EdTech   
3         Bizongo   2015.0      Mumbai  B2B E-commerce   
4        FypMoney   2021.0    Gurugram         FinTech   

                                        What it does   
0  Unbox Robotics builds on-demand AI-driven ware...  \
1     UpGrad is an online higher education platform.   
2  LEAD School offers technology based school tra...   
3  Bizongo is a business-to-business online marke...   
4  FypMoney is Digital NEO Bank for Teenagers, em...   

                                            Founders   
0                        Pramod Ghadge, Shahid Memon  \
1  Mayank Kumar, Phalgun Kompalli, Ravijot Chugh,...   
2                         Smita Deorah, Sumeet Mehta   
3            Aniket Deb, Ankit Tomar, Sachin Agrawal   
4                                      Kapil Banwari   

                                        Investor     Amount($)         Stage  
0                    BEENEXT, Entrepreneur First    $1,200,000  Pre-series A  
1       Unilazer Ventures, IIFL Asset Management  $120,000,000           NaN  
2               GSV Ventures, Westbridge Capital   $30,000,000      Series D  
3                         CDC Group, IDG Capital   $51,000,000      Series C  
4  Liberatha Kallat, Mukesh Yadav, Dinesh Nagpal    $2,000,000          Seed

In [283]:
# Dropping the "Founded", "Founders" and "Investor" columns

df4 = df4.drop(["Founded", "Founders", "Investor"], axis =1)
df4.head()

Company/Brand HeadQuarter          Sector   
0  Unbox Robotics   Bangalore      AI startup  \
1          upGrad      Mumbai          EdTech   
2     Lead School      Mumbai          EdTech   
3         Bizongo      Mumbai  B2B E-commerce   
4        FypMoney    Gurugram         FinTech   

                                        What it does     Amount($)   
0  Unbox Robotics builds on-demand AI-driven ware...    $1,200,000  \
1     UpGrad is an online higher education platform.  $120,000,000   
2  LEAD School offers technology based school tra...   $30,000,000   
3  Bizongo is a business-to-business online marke...   $51,000,000   
4  FypMoney is Digital NEO Bank for Teenagers, em...    $2,000,000   

          Stage  
0  Pre-series A  
1           NaN  
2      Series D  
3      Series C  
4          Seed

In [284]:
df4.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1209 entries, 0 to 1208
Data columns (total 6 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   Company/Brand  1209 non-null   object
 1   HeadQuarter    1208 non-null   object
 2   Sector         1209 non-null   object
 3   What it does   1209 non-null   object
 4   Amount($)      1206 non-null   object
 5   Stage          781 non-null    object
dtypes: object(6)
memory usage: 56.8+ KB


In [285]:
# Checking for missing values

df4.isna().sum()

Company/Brand      0
HeadQuarter        1
Sector             0
What it does       0
Amount($)          3
Stage            428
dtype: int64

In [286]:
# Checking the total number of rows

df4.shape

(1209, 6)

Since there are 1209 rows in the dataset, we can drop the missing values in the "HeadQuarter" and "Amount ($)" columns since they are little, and only have to fill the missing values in the "Stage" column since they are much.

In [287]:
# Filling the missing values using backward fill method

df4["Stage"] = df4["Stage"].fillna(method="bfill")

df4.isna().sum()

Company/Brand    0
HeadQuarter      1
Sector           0
What it does     0
Amount($)        3
Stage            0
dtype: int64

In [290]:
# Dropping the remaining missing values

df4 = df4.dropna()

df4.isna().sum()

Company/Brand    0
HeadQuarter      0
Sector           0
What it does     0
Amount($)        0
Stage            0
dtype: int64

In [291]:
df4.head()

Company/Brand HeadQuarter          Sector   
0  Unbox Robotics   Bangalore      AI startup  \
1          upGrad      Mumbai          EdTech   
2     Lead School      Mumbai          EdTech   
3         Bizongo      Mumbai  B2B E-commerce   
4        FypMoney    Gurugram         FinTech   

                                        What it does     Amount($)   
0  Unbox Robotics builds on-demand AI-driven ware...    $1,200,000  \
1     UpGrad is an online higher education platform.  $120,000,000   
2  LEAD School offers technology based school tra...   $30,000,000   
3  Bizongo is a business-to-business online marke...   $51,000,000   
4  FypMoney is Digital NEO Bank for Teenagers, em...    $2,000,000   

          Stage  
0  Pre-series A  
1      Series D  
2      Series D  
3      Series C  
4          Seed

In [292]:
# Checking for duplicate rows

df4.duplicated().sum()

18

The dataset has eighteen duplicate rows

In [293]:
# Dropping the duplicate row

df4 = df4.drop_duplicates()

# Confirming that the duplicate row has been dropped

df4.duplicated().sum()

0

In [294]:
# Checking for custom quality

df4["Amount($)"].apply(type).value_counts()

Amount($)
<class 'str'>    1187
Name: count, dtype: int64

In [295]:
# Removing the dollar and commma symbols in the entire Amount($) column

df4['Amount($)'] = df4['Amount($)'].replace(['\$', ','], '', regex=True)
df4.head()

Company/Brand HeadQuarter          Sector   
0  Unbox Robotics   Bangalore      AI startup  \
1          upGrad      Mumbai          EdTech   
2     Lead School      Mumbai          EdTech   
3         Bizongo      Mumbai  B2B E-commerce   
4        FypMoney    Gurugram         FinTech   

                                        What it does  Amount($)         Stage  
0  Unbox Robotics builds on-demand AI-driven ware...    1200000  Pre-series A  
1     UpGrad is an online higher education platform.  120000000      Series D  
2  LEAD School offers technology based school tra...   30000000      Series D  
3  Bizongo is a business-to-business online marke...   51000000      Series C  
4  FypMoney is Digital NEO Bank for Teenagers, em...    2000000          Seed

In [296]:
# Creating a new column named 'Year' and filling it with 2021, which is the dataset year

df4['Year'] = 2021
df4.head()

Company/Brand HeadQuarter          Sector   
0  Unbox Robotics   Bangalore      AI startup  \
1          upGrad      Mumbai          EdTech   
2     Lead School      Mumbai          EdTech   
3         Bizongo      Mumbai  B2B E-commerce   
4        FypMoney    Gurugram         FinTech   

                                        What it does  Amount($)         Stage   
0  Unbox Robotics builds on-demand AI-driven ware...    1200000  Pre-series A  \
1     UpGrad is an online higher education platform.  120000000      Series D   
2  LEAD School offers technology based school tra...   30000000      Series D   
3  Bizongo is a business-to-business online marke...   51000000      Series C   
4  FypMoney is Digital NEO Bank for Teenagers, em...    2000000          Seed   

   Year  
0  2021  
1  2021  
2  2021  
3  2021  
4  2021

# Things to be done

1. Rectify the challenge in the "Sector" column of the first dataset
2. Save each of the cleaned datasets into new csv files
3. Merge each of the cleaned datasets into one csv file
4. EDA
5. Hypothesis testing
6. Visualizations